### PR TITLE
ref(app-platform): UI for internal app details

### PR DESCRIPTION
**Pull request by MeredithAnya**
This PR adds the first UI pieces for Internal Apps.

**What are internal apps?**
In short, they are a simplified version of the current applications you can build on the Sentry Integration Platform.
* No OAuth process needed
* Automatically installed and given an access token (which lasts forever unless regenerated)
* Only scoped to the Sentry organization where the application was created

**UI Changes** 
Toggle for making an app internal 
>Screen Shot 2019-05-20 at 10 54 14 AM

Disable when already internal
Screen Shot 2019-05-21 at 1 44 29 PM

Installation ID and token are the credentials shown

### CODE REVIEW

File: src/sentry/static/sentry/app/data/forms/sentryApplication.jsx [hunk 1]
```diff
@@ -6,6 +6,12 @@ const forms = [
     // Form "section"/"panel"
     title: 'Integration Details',
     fields: [
+      {
+        name: 'internal',
+        label: 'Internal',
+        type: 'boolean',
+        help: 'Choose this to make your integration Internal.',
```
Comment (markstory) [inline src/sentry/static/sentry/app/data/forms/sentryApplication.jsx]: What does `internal` mean?

Comment (MeredithAnya) [reply src/sentry/static/sentry/app/data/forms/sentryApplication.jsx]: tried to add better wording but I do think we'll have to link to docs to explain it more. 
tl;dr is there is no oauth process, they just get an access token, and there is no installing or uninstalling. they can just use the token (and regenerate it if need be)

Comment (mnoble) [reply src/sentry/static/sentry/app/data/forms/sentryApplication.jsx]: Maybe it'd be more descriptive as "Skips OAuth and only accessible in this Org" or something concise that describes what you did above?

File: src/sentry/static/sentry/app/data/forms/sentryApplication.jsx [hunk 2]
```diff
@@ -6,6 +6,13 @@ const forms = [
     // Form "section"/"panel"
     title: 'Integration Details',
     fields: [
+      {
+        name: 'internal',
+        label: 'Internal',
+        type: 'boolean',
+        help:
+          'If enabled, your integration will automatically be installed and for use within your organization only.',
```
Comment (markstory) [inline src/sentry/static/sentry/app/data/forms/sentryApplication.jsx]:
```suggestion
          'If enabled, your integration will automatically be installed for use within only your organization.',
```

File: src/sentry/static/sentry/app/data/forms/sentryApplication.jsx (hunk 1 above)
Comment (MeredithAnya) [reply src/sentry/static/sentry/app/data/forms/sentryApplication.jsx]: personally, I think saying it's automatically installed says more about whats happening than skips OAuth.

**Comment by markstory** [comment]
Looks good to me.

File: src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.jsx [hunk 3]
```diff
@@ -119,28 +120,51 @@ export default class SentryApplicationDetails extends AsyncView {
...
+                          {getDynamicText({value, fixed: 'PERCY_CLIENT_ID'})}
+                        </TextCopyInput>
+                      );
+                    }}
+                  </FormField>
+                  <FormField overflow name="installation" label="Installation ID">
+                    {({value}) => {
+                      return (
+                        <TextCopyInput>
+                          {getDynamicText({value: value.uuid, fixed: 'PERCY_CLIENT_ID'})}
```
Comment (mnoble) [inline src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.jsx]: We should probably change the `fixed` values here and above, for clarity.

**Review by mnoble — CHANGES_REQUESTED** [review]
We should adds tests to ensure the endpoint (serializer) exposes the correct data. That branch in the serializer is important.

**Review by mnoble — APPROVED** [review]
:shipit:
